docs/source/extension/extension_points.rst

```diff
@@ -457,16 +457,180 @@ The recommended ranges for this rank are:
 Main Menu
 ---------
 
-There are three main ways to extend JupyterLab's main menu.
+There are two ways to extend  JupyterLab's main menu.
+
+1. Using the settings - this is the preferred way as they are configurable by the user.
+2. Using the API - this is for advanced cases like dynamic menu or semantic items.
+
+Settings-defined menu
+^^^^^^^^^^^^^^^^^^^^^
+
+JupyterLab provides integration with its settings system for menu definitions.
+Your extension can provide a settings schema with a ``jupyter.lab.menus`` key,
+declaring default menus. You don't need to set anything in the typescript code
+(except the command definitions).
+
+To add a new menu with your extension command:
+
+.. code:: json
+
+    {
+      "jupyter.lab.menus": {
+      "main": [
+        {
+          "id": "jp-mainmenu-myextension",
+          "label": "My Menu",
+          "items": [
+            {
+              "command": "my-command",
+              "rank": 500
+            }
+          ],
+          "rank": 100
+        }
+      ]
+    }
+
+The menu item label will be set with the command label. For menus (and
+submenus), the label needs to be set explicitly with the ``label`` 
+property.
+
+Menu and item have a ``rank`` that will determine the elements order.
+
+
+To add a new entry in an existing menu:
+
+.. code:: json
+
+    {
+      "jupyter.lab.menus": {
+      "main": [
+        {
+          "id": "jp-mainmenu-file",
+          "items": [
+            {
+              "command": "my-command",
+              "rank": 500
+            }
+          ]
+        }
+      ]
+    }
+
+Here is the list of default menu ids:
+
+- File menu: ``jp-mainmenu-file``
+
+  * New file submenu: ``jp-mainmenu-file-new``
+  
+- Edit menu: ``jp-mainmenu-edit``
+- View menu: ``jp-mainmenu-view``
+- Run menu: ``jp-mainmenu-run``
+- Kernel menu: ``jp-mainmenu-kernel``
+- Tabs menu: ``jp-mainmenu-tabs``
+- Settings menu: ``jp-mainmenu-settings``
+- Help menu: ``jp-mainmenu-help``
+
+The default main menu is defined in the ``mainmenu-extension`` package settings.
+
+A menu must respect the following schema:
+
+.. code:: json
+
+    "menu": {
```

Comment:
mmm... delicious. i wonder if we can pull it directly from the file instead of reproducing it here? i'm not much of an rst-wizard...

Reply:
Done